src/storage_manifests.py

```diff
@@ -145,6 +155,51 @@ def evaluate(self) -> Optional[str]:
                 return f"Storage manifests waiting for definition of {prop}"
         return None
 
+    def is_ready(self, obj: HashableResource, cond: AnyCondition) -> Optional[bool]:
+        """Determine if the resource is ready."""
+        if not super().is_ready(obj, cond):
+            object_events = collect_events(self.client, obj.resource)
+            if obj.kind in ["Deployment", "DaemonSet"]:
+                involved_pods = self.client.list(
+                    Pod, namespace=obj.namespace, labels={"app": obj.name}
+                )
+                object_events += [
+                    event for pod in involved_pods for event in collect_events(self.client, pod)
+                ]
+            for event in sorted(object_events, key=by_localtime):
+                log.info(
+                    "Event %s/%s %s msg=%s",
+                    event.involvedObject.kind,
+                    event.involvedObject.name,
+                    event.lastTimestamp
+                    and event.lastTimestamp.astimezone()
+                    or "Date not recorded",
+                    event.message,
+                )
+
+        return super().is_ready(obj, cond)
+
+
+def by_localtime(event: Event) -> datetime.datetime:
+    """Return the last timestamp of the event in local time."""
+    dt = event.lastTimestamp or datetime.datetime.now(datetime.timezone.utc)
+    return dt.astimezone()
+
+
+def collect_events(client, resource: AnyResource) -> List[Event]:
+    """Collect events from the resource."""
+    kind: str = resource.kind or type(resource).__name__
+    return list(
+        client.list(
+            Event,
+            namespace=resource.metadata.namespace,
+            fields={
+                "involvedObject.kind": kind,
+                "involvedObject.name": resource.metadata.name,
+            },
+        )
+    )
```

Comment:
Is this a list of lists?

Reply:
[lightkube `client.list` ](https://lightkube.readthedocs.io/en/stable/client/) returns an iterator.  So this is actually building an ACTUAL python list of events from the returned python iterator. 

iterators are something you can call `next(it)` on until it yields a `StopIteration` exception. 

so by running `list(it)` you get an in-memory list version of that iterated item.  similar to `list(range(10)` providing you [0,1,2,3,4,5,6,7,8,9]`.  the `range(...)` method yields an iterator